modules/standard/FileSystem.chpl

```diff
@@ -1098,8 +1098,10 @@ iter listdir(path: string = ".", hidden: bool = false, dirs: bool = true,
 */
 iter listDir(path: string = ".", hidden: bool = false, dirs: bool = true,
               files: bool = true, listlinks: bool = true): string {
-  extern type DIRptr = c_ptr(opaque);
-  extern type direntptr = c_ptr(opaque);
+  extern record DIR {};
+  extern type DIRptr = c_ptr(DIR);
+  extern "struct dirent" record chpl_dirent {};
```

Comment:
Not necessary to change, but for general reference, not that Chapel class / record / enum declarations do not require a terminating `;`.  In essence, this is just a no-op statement following the declaration, like:

```chapel
extern record DIR {
} // record declaration stops here
; // new statement here
```